File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.java
import java.lang.*;
class Solution {
    public int solution(int n) {
        int answer = 0;
        for(int i=2;i<=n;i++){
            int k=(int)Math.sqrt(i);
            boolean isOdd=true;
            for(int j=2;j<=k;j++){
                if (i%j==0){
                    isOdd=false; 
                    break;
                }
            }
            if (isOdd) answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
import math
from collections import deque

numset = set()
answer = 0
def solution(numbers):
    digits = list(numbers)
    visited = [False for _ in range(len(digits))]
     #나온 숫자
    for i in range(len(digits)):
        addnum(i,digits[i],digits,visited)
    global answer
    return answer
    
def addnum(idx, num, digits, visited):
    visited[idx] = True
    global numset
    if int(num) not in numset:
        if checksosu(int(num)):
            global answer
            answer+=1
        numset.add(int(num))
    for next in range(len(digits)):
        if not visited[next]:
            addnum(next, str(num)+digits[next], digits, visited)
            visited[next]=False
    visited[idx] = False # backtracking
    return
    
def checksosu(num):
    if num<=1: return False
    for i in range(2, int(math.sqrt(num))+1):
        if num%i==0: return False
    return True
    